backend/deepchecks_monitoring/schema_models/permission_mixin.py

```diff
@@ -0,0 +1,52 @@
+# ----------------------------------------------------------------------------
+# Copyright (C) 2021-2022 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+# pylint: disable=ungrouped-imports,import-outside-toplevel
+"""Module defining utility functions for the deepchecks_monitoring app."""
+import abc
+
+import fastapi
+import sqlalchemy as sa
+from fastapi import Depends
+
+from deepchecks_monitoring.dependencies import AsyncSessionDep
+from deepchecks_monitoring.exceptions import AccessForbidden, NotFound
+
+
+class PermissionMixin:
+    """Mixin class for ORM entities that have metadata."""
+
+    from deepchecks_monitoring.public_models.user import User
+    from deepchecks_monitoring.utils import auth
+
+    @classmethod
+    @abc.abstractmethod
+    def get_object_by_id(cls, id, user):
+        raise NotImplementedError()
+
+    @classmethod
+    async def fetch_or_403(cls, session, id, user):
+        obj = await session.scalar(cls.get_object_by_id(id, user))
+        if obj is None:
+            raise AccessForbidden(f"You do not have permissions to access this object.")
+        return obj
+
+    @classmethod
+    async def get_object(cls, request: fastapi.Request,
```

Comment:
do not pass request instance here, instead pass id directly

If wanted to use this function in some different context, not in an HTTP handler, I would have to create an 'artificial' request object/